sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs

```diff
@@ -168,26 +169,51 @@ private async Task<int> DownloadInternal(bool async, CancellationToken cancellat
 
             using Stream networkStream = response.Value.Content;
 
+            // The number of bytes we just downloaded.
+            long downloadSize = GetResponseRange(response.GetRawResponse()).Length.Value;
+
+            // The number of bytes we copied in the last loop.
             int copiedBytes;
 
+            // Bytes we have copied so far.
+            int totalCopiedBytes = 0;
+
+            // Bytes remaining to copy.
+            int remainingBytes = (int)downloadSize;
+
             if (async)
```

Comment:
Nit - consider moving the async check inside the loop to avoid doubling the logic.

Reply:
Fixed.